src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -111,35 +116,29 @@ public void periodic() {
         boolean stopOpening = !allowOpen && prevAllowedOpen; // only heed falling edge of allowed open so we can open from ground position
         prevAllowedOpen = allowOpen;
 
-        boolean opening = openTimer.get() > 0 && !openTimer.hasElapsed(OPEN_TIME_SECONDS);
-        boolean closing = !opening || stopOpening;
-
-        if (closing) {
+        // If we're opening and not allowed to anymore, or if we've finished opening, stop opening and start closing.
+        if ((openTimer.hasStarted() && stopOpening) || openTimer.hasElapsed(OPEN_TIME_SECONDS)) {
+            openTimer.stop();
+            openTimer.reset();
             closeTimer.start();
+            cooldownTimer.start();
```

Comment:
i do agree it makes sense to start the cooldown after we stop opening, but i think the old logic included opening time. can we adjust the cooldown constant to match?

Reply:
I thought the old logic didn't though; https://github.com/grt192/GRT2023/pull/61/files#diff-412251c91fef77914d9fc678673a89d2d22664ea9c1a90b42e59bb0bdcd98745L131-L132

Reply:
oh okay you're right